### PR TITLE
[WIP] scvis

### PR DESCRIPTION
**Pull request by astrovsky01**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/scvis/scvis.xml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+<tool id="scvis_train" name="SCVIS train" version="0.1.0">
+    <description>Parametric alternative to t-SNE for use in high-dimensional data analysis.</description>
+    <requirements>
+        <requirement type="package" version="0.1.0">scvis</requirement>
+        <requirement type="package">tensorflow</requirement>
```
Comment (bgruening) [inline tools/scvis/scvis.xml]: can you add here a version?

File: tools/scvis/scvis.xml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      --config_file '${configfile}'
+  #end if
+  #if $norm.normalize == "Yes":
+      --normalize your_number '${normval}'
+  #end if
+  --out_dir .
+  --show_plot
+    ]]></command>
+    <inputs>
+        <param name="datafile" type="data" label="Data Matrix File" help="Data matrix with first row as the column names" format="tsv"/>
```
Comment (bgruening) [inline tools/scvis/scvis.xml]: just for the style, can you shift the type before the label and help, see here: https://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html#coding-style

File: tools/scvis/scvis.xml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            <conditional name="labels">
+                <param name="label" type="select" label="Provide clustering information" help="Give cluster information for graph coloring?">
+                    <option value="No">No</option>
+                    <option value="Yes">Yes</option>
+                </param>
+                <when value="No"/>
+                <when value="Yes">
+                    <param name="datalabel" type="data" label="Data Label File" help="Optional: One column file with header provinding corresponding cluster information for each data point. Used for coloring plots." format="tsv"/>
```
Comment (bgruening) [inline tools/scvis/scvis.xml]: you can remove this conditional and simply use optional="true" on the dataset ...

File: tools/scvis/scvis.xml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                </when>
+            </conditional>
+            <conditional name="config">
+                <param name="choice" type="select" label="Provide config file">
+                    <option value="No">No</option>
+                    <option value="Yes">Yes</option>
+                </param>
+                <when value="No"/>
+                <when value="Yes">
+                    <param name="configfile" type="data" label="Config File" help="Optional: Config yaml file as specified in the scvis docs" format="yaml" optional="true"/>
```
Comment (bgruening) [inline tools/scvis/scvis.xml]: same here ...

File: tools/scvis/scvis.xml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                </when>
+            </conditional>
+            <conditional name="norm">
+                <param name="normalize" type="select" label="Set Normalization Value?" help="If not selected, normalization will use maximum absolute value">
+                    <option value="No">No</option>
+                    <option value="Yes">Yes</option>
+                </param>
+                <when value="No"/>
+                    <param name="normval" value=""/>
```
Comment (bgruening) [inline tools/scvis/scvis.xml]: this parameter is missing a type I think.

File: tools/scvis/scvis.xml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+                <when value="Yes">
+                     <param name="normval" type="float" label="Normalization value" help="A positive float" value="0.5"/>
+                </when>
+            </conditional>
+    </inputs>
+    <outputs>
+      <collection name="graphs" type="list" label="Graphical_outputs">
+          <discover_datasets pattern=".png" directory="outputs/*_result/embedding_iter_00000"/>
```
Comment (bgruening) [inline tools/scvis/scvis.xml]: this pattern is not working isn't it?

File: tools/scvis/scvis.xml (hunk 1 above)
Comment (astrovsky01) [reply tools/scvis/scvis.xml]: That was me trying to import the default version. Now that I know it's not an error, I can add the version back in

File: tools/scvis/scvis.xml (hunk 3 above)
Comment (astrovsky01) [reply tools/scvis/scvis.xml]: When I did that, Galaxy wasn't letting me not select a file.

File: tools/scvis/scvis.xml (hunk 6 above)
Comment (astrovsky01) [reply tools/scvis/scvis.xml]: No, it isn't. That's the last big hurdle right now: identifying outputs.

File: tools/scvis/scvis.xml (hunk 3 above)
Comment (bgruening) [reply tools/scvis/scvis.xml]: Did you used optional="true"? I'm sure this is working. Otherwise, its a bug.

**Pull request closed without merging**